File: src/Roguelike.Core/Core/Room/BossRoomHandler.cs
using Roguelike.Data;
using Roguelike.Core.Map;
using System.Linq;

namespace Roguelike.Core.Handlers
{
    public class BossRoomHandler : IRoomHandler
    {
        public void Execute(GameRun run, Room room)
        {
            
            var boss = run.EnemyPool.GetRandomEnemyOfStar(5, run.Rng);
            if (boss == null) return;

            var encounter = new System.Collections.Generic.List<EnemyData> { boss };

            run.CurrentCombat = new CombatManager(run.TheHero, encounter, run.Rng, run.EffectPool.GetEffect);
            run.CurrentCombat.StartCombat();
            run.CurrentState = GameState.InCombat;
        }
    }
}

File: python_roguelike/core/rooms/boss_room_handler.py
from .i_room_handler import IRoomHandler
from ..combat_manager import CombatManager
from ...data.enums import GameState


class BossRoomHandler(IRoomHandler):
    def execute(self, run, room):
        boss = run.enemy_pool.get_random_enemy_of_star(5, run.rng)
        if boss is None:
            return

        run.current_combat = CombatManager(
            run.the_hero,
            [boss],
            run.rng,
            run.effect_pool.get_effect
        )
        run.current_combat.start_combat()
        run.current_state = GameState.InCombat
